File: app/src/main/java/vergecurrency/vergewallet/service/model/network/layers/ClearnetGateway.java
package vergecurrency.vergewallet.service.model.network.layers;

import android.os.AsyncTask;

import cz.msebera.android.httpclient.HttpEntity;
import cz.msebera.android.httpclient.HttpResponse;
import cz.msebera.android.httpclient.client.HttpClient;
import cz.msebera.android.httpclient.client.methods.HttpGet;
import cz.msebera.android.httpclient.impl.client.DefaultHttpClient;
import cz.msebera.android.httpclient.util.EntityUtils;

public class ClearnetGateway extends AsyncTask<String, Void, String> {


	public ClearnetGateway() {
	}


	@Override
	protected String doInBackground(String... url) {
		try {


			//TODO : CHECK INTERNET CONNECTION YOU IDIOT
			HttpGet httppost = new HttpGet(url[0]);
			HttpClient httpclient = new DefaultHttpClient();
			HttpResponse response = httpclient.execute(httppost);

			// StatusLine stat = response.getStatusLine();
			int status = response.getStatusLine().getStatusCode();

			if (status == 200) {
				HttpEntity entity = response.getEntity();
				String data = EntityUtils.toString(entity);


				return data;
			} else return "error";

		} catch (Exception e) {
			e.printStackTrace();
			return null;
		}

	}
}

File: app/src/main/kotlin/vergecurrency/vergewallet/service/model/network/layers/ClearnetGateway.kt
package vergecurrency.vergewallet.service.model.network.layers

import android.os.AsyncTask
import cz.msebera.android.httpclient.client.methods.*
import cz.msebera.android.httpclient.entity.StringEntity
import cz.msebera.android.httpclient.impl.client.CloseableHttpClient
import cz.msebera.android.httpclient.impl.client.HttpClientBuilder
import cz.msebera.android.httpclient.util.EntityUtils


class ClearnetGateway(requestBase: HttpRequestBase) : Gateway(requestBase) {
    override fun background(vararg params: String?): String {
        try {
            val httpClient = HttpClientBuilder.create().build()
            val response =  httpClient.execute(super.requestBase);
            val status = response.statusLine.statusCode

            if (status == 200) {
                val entity = response.entity


                return EntityUtils.toString(entity)
            } else
                return ""

        } catch (e: Exception) {
            //TODO
            e.printStackTrace()
            return "";
        }
    }


}
